llvm/test/CodeGen/CHERI-Generic/Inputs/atomic-rmw-cap-ptr-cap-arg.ll

```diff
@@ -0,0 +1,142 @@
+@IF-RISCV@; RUN: %generic_cheri_hybrid_llc %s -o - -mattr=+f,+a -verify-machineinstrs | FileCheck %s --check-prefixes=HYBRID,HYBRID-ATOMICS
+@IF-RISCV@; RUN: %generic_cheri_hybrid_llc %s -o - -mattr=+f,-a -verify-machineinstrs | FileCheck %s --check-prefixes=HYBRID,HYBRID-LIBCALLS
+@IFNOT-RISCV@; RUN: %generic_cheri_hybrid_llc %s -o - | FileCheck %s --check-prefix=HYBRID
+
+define dso_local i8 addrspace(200)* @atomic_cap_ptr_xchg_sc(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val seq_cst
+  ret i8 addrspace(200)* %tmp
+}
+
+define dso_local i8 addrspace(200)* @atomic_cap_ptr_xchg_relaxed(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val monotonic
+  ret i8 addrspace(200)* %tmp
+}
+
+define dso_local i8 addrspace(200)* @atomic_cap_ptr_xchg_acquire(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val acquire
+  ret i8 addrspace(200)* %tmp
+}
+
+define dso_local i8 addrspace(200)* @atomic_cap_ptr_xchg_rel(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val release
+  ret i8 addrspace(200)* %tmp
+}
+
+define dso_local i8 addrspace(200)* @atomic_cap_ptr_xchg_acq_rel(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %val acq_rel
+  ret i8 addrspace(200)* %tmp
+}
+
+; Also check non-i8* xchg:
+define dso_local i32 addrspace(200)* @atomic_cap_ptr_xchg_i32ptr(i32 addrspace(200)* addrspace(200)* %ptr, i32 addrspace(200)* %val) nounwind {
+bb:
+  %tmp = atomicrmw xchg i32 addrspace(200)* addrspace(200)* %ptr, i32 addrspace(200)* %val acq_rel
+  ret i32 addrspace(200)* %tmp
+}
+
+; Function Attrs: nofree norecurse nounwind
+define zeroext i1 @test_cmpxchg_weak(i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* addrspace(200)* nocapture %exp, i8 addrspace(200)* %new) local_unnamed_addr #0 {
+entry:
+  %0 = load i8 addrspace(200)*, i8 addrspace(200)* addrspace(200)* %exp, align 16
+  %1 = cmpxchg weak i8 addrspace(200)* addrspace(200)* %ptr, i8 addrspace(200)* %0, i8 addrspace(200)* %new acq_rel acquire
+  %2 = extractvalue { i8 addrspace(200)*, i1 } %1, 1
+  br i1 %2, label %cmpxchg.continue, label %cmpxchg.store_expected
+
+cmpxchg.store_expected:                           ; preds = %entry
+  %3 = extractvalue { i8 addrspace(200)*, i1 } %1, 0
+  store i8 addrspace(200)* %3, i8 addrspace(200)* addrspace(200)* %exp, align 16
+  br label %cmpxchg.continue
+
+cmpxchg.continue:                                 ; preds = %cmpxchg.store_expected, %entry
+  ret i1 %2
```

Comment:
```suggestion
  ret void
```
just like llvm/test/CodeGen/RISCV/atomic-cmpxchg.ll, no need for all the other stuff.